crates/ra_hir_expand/src/name.rs

```diff
@@ -61,9 +120,40 @@ impl Name {
         Name::new_text("[missing name]".into())
     }
 
+    fn from_repr(repr: Repr) -> Self {
+        /*
+        Encoding:
+                 0               reserved
+                 1 - 0x7fffffff  interned string
+        0x80000000 - 0xffffffff  tuple field
+        */
+
+        match repr {
+            Repr::Interned(spur) => {
+                // Seems to be unsafe by mistake.
+                let spur = unsafe { spur.into_usize() } as u32;
```

Comment:
`Key::into_usize` is unsafe because there's internal [safety guarantees](https://docs.rs/lasso/0.3.0/lasso/trait.Key.html#safety-1) on it. For your use case, I would actually recommend making a custom implementation of `Key` that enforces your `1 - 0x7fffffff` rule ([example impl](https://github.com/Kixiron/lasso/blob/master/src/key.rs#L39-L60))

Reply:
Those seem to be constraints on `Key` implementors only, not on callers, right?

Reply:
Yes, it was originally meant to press the importance of upholding them, but multiple people have mentioned this so I'll change it